tests/tqueue_int/tqueue_int.c

```diff
@@ -335,21 +361,29 @@ static int pusher_thread_func(void* arg)
     return 0;
 }
 
-static int popper_thread_func(void* arg)
+static int popper_threads_func(void* arg)
 {
-    ONE_PUSHER_ONE_POPPER_TEST_CONTEXT* test_context = arg;
+    TQUEUE_CHAOS_TEST_THANDLE_CONTEXT* test_context = arg;
 
-    while (interlocked_add(&test_context->terminate_test, 0) == 0)
+    // Checking timeout locally in each thread since yielding of CPU to main thread may be delayed or not materialized
+    // fast enough due to high number of threads doing intensive work. Due to this the ctest timeout of 1500 sec may get
+    // triggered and kill the test
+    double start_time = timer_global_get_elapsed_ms();
+    double current_time = start_time;
+    while ((interlocked_add(&test_context->terminate_test, 0) == 0) && (current_time - start_time < CHAOS_TEST_RUNTIME))
     {
-        FOO item = { .x = -1 };
-        TQUEUE_POP_RESULT pop_result = TQUEUE_POP(FOO)(test_context->queue, &item, NULL, NULL, NULL);
-        ASSERT_IS_TRUE((pop_result == TQUEUE_POP_OK) || (pop_result == TQUEUE_POP_QUEUE_EMPTY), "TQUEUE_POP(FOO) failed with %" PRI_MU_ENUM "", MU_ENUM_VALUE(TQUEUE_POP_RESULT, pop_result));
+        THANDLE(TEST_THANDLE) item = NULL;
+        TQUEUE_POP_RESULT pop_result = TQUEUE_POP(THANDLE(TEST_THANDLE))(test_context->queue, &item, NULL, NULL, NULL);
+        ASSERT_IS_TRUE((pop_result == TQUEUE_POP_OK) || (pop_result == TQUEUE_POP_QUEUE_EMPTY), "TQUEUE_POP(THANDLE(TEST_THANDLE)) failed with %" PRI_MU_ENUM "", MU_ENUM_VALUE(TQUEUE_POP_RESULT, pop_result));
         if (pop_result == TQUEUE_POP_OK)
         {
-            int64_t expected_pop_number = interlocked_increment_64(&test_context->next_expected_pop_number) - 1;
-            ASSERT_ARE_EQUAL(int, expected_pop_number, item.x);
+            (void)interlocked_increment_64(&test_context->successful_pop_count);
```

Comment:
>(void)interlocked_increment_64(&test_context->successful_pop_count); [](http://example.com/codeflow?start=12&length=68)

So I know we can't validate the "next expected" number when there are multiple threads, but we should at least make sure we received a non-NULL `item` and maybe keep the check that the "x" is not "-1".